File: Sample/Vibes SDK Objective-C Sample App/InboxDetailViewController.m
//
//  InboxDetailViewController.m
//  Vibes SDK Objective-C Sample App
//
//  Created by Moin' Victor on 25/11/2019.
//  Copyright © 2019 Vibes. All rights reserved.
//

#import "InboxDetailViewController.h"

@interface InboxDetailViewController ()

@end

@implementation InboxDetailViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self.contentLabel setText: _inboxMessage.content];
    [self.subjectLabel setText: _inboxMessage.subject];
    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
    dateFormatter.dateFormat = @"yyyy-MM-dd 'at' HH:mm:ss";
    NSString *dateString = [dateFormatter stringFromDate: _inboxMessage.createdAt];
    [self.dateLabel setText:dateString];

    [self downloadImageFromURL:_inboxMessage.detail intoImageView:self.imageIcon withActivityIndicator:self.activityIndicator];
    // Do any additional setup after loading the view.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

- (void) downloadImageFromURL: (NSString*) url intoImageView: (UIImageView*) imageView
        withActivityIndicator: (UIActivityIndicatorView*) activityIndicator
{
    [activityIndicator startAnimating];
    [activityIndicator setHidden:NO];
    dispatch_async(dispatch_get_global_queue(0,0), ^{
        NSData * data = [[NSData alloc] initWithContentsOfURL: [NSURL URLWithString: url]];
        if ( data == nil) {
            dispatch_async(dispatch_get_main_queue(), ^{
                [activityIndicator stopAnimating];
                [activityIndicator setHidden:YES];
            });
            return;
        }
        dispatch_async(dispatch_get_main_queue(), ^{
            // WARNING: is the cell still using the same data by this point??
            imageView.image = [UIImage imageWithData: data];
            [activityIndicator stopAnimating];
            [activityIndicator setHidden:YES];
        });
    });
}
@end

File: SampleSwift/VibesSDKSampleApp/InboxDetailViewController.swift
//
//  InboxDetailViewController.swift
//  VibesSDKSampleApp
//
//  Created by Moin' Victor on 22/11/2019.
//  Copyright © 2019 Vibes Media. All rights reserved.
//

import UIKit
import VibesPush

/// Inbox Detail
class InboxDetailViewController: UIViewController {

    var inboxMessage: InboxMessage?

    @IBOutlet weak var contentLabel: UILabel!
    @IBOutlet weak var dateLabel: UILabel!
    @IBOutlet weak var imageIcon: UIImageView!
    @IBOutlet weak var subjectLabel: UILabel!
    @IBOutlet weak var activityIndicator: UIActivityIndicatorView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        if let inboxMessage = inboxMessage {
            self.subjectLabel.text = inboxMessage.subject
            self.contentLabel.text = inboxMessage.content
            let date = inboxMessage.createdAt
            let dateFormat = DateFormatter()
            dateFormat.dateFormat = "yyyy-MM-dd hh:mm:ss"
            dateLabel.text = dateFormat.string(from: date)
            if let detail = inboxMessage.detail {
                // download image, you can use any library here
                downloadImage(from: URL(string: detail)!, into: self.imageIcon, withIndicator: activityIndicator)
            }
        }
    }

}
